builder/azure/arm/builder.go

```diff
@@ -82,11 +83,21 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 		return nil, err
 	}
 
+	resolver := newResourceResolver(azureClient)
+	if err := resolver.Resolve(b.config); err != nil {
+		return nil, err
+	}
+
 	b.config.storageAccountBlobEndpoint, err = b.getBlobEndpoint(azureClient, b.config.ResourceGroupName, b.config.StorageAccount)
 	if err != nil {
 		return nil, err
 	}
 
+	endpointConnectType := PublicEndpoint
```

Comment:
You can provide ssh_host argument of your ssh communicator with the private IP address whilst connected to VPN and should work without need for a public IP.